### DISCUSSION TITLE
Support for bundle shape for PDE maven targets

### DISCUSSION THREAD

**Discussion by joeshannon** [eclipse-m2e/m2e-core#1497] [Ideas]
Whilst switching some dependencies to be sourced from Maven instead of p2 repositories I have found a limitation if the dependency contains `Eclipse-BundleShape: dir`. This is used for cases where the dependency should be installed in directory form instead of jar - e.g. for native libraries or resources.

When sourced from a p2 site, these bundles are installed as directories (in `<workspace>/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool`). However when the same bundles are consumed from Maven the jar within `~/.m2/repository/..` is used directly.

Would it be possible to support this? Perhaps the dependencies could be handled in a similar way to how wrapped (non-bundle) maven dependencies are handled by m2e pde?

**Comment by laeubi** [comment]
I don't see a reason to not support it, one could simply extract the jar in the current folder (e.g. `eclipse-dir-bundle`) and then use this folder instead, do you like to propose a PR for this? As we already know the manifest of the bundles it shouldn't be too hard.

**Reply by joeshannon** [reply]
Yes I can try and make a PR.
I've had a quick look at the code just wanted a few pointers as to where would be the best place to do this.
Is `MavenTargetBundle` the correct place to be looking? This constructor contains the logic for deciding whether the maven artifact needs to be wrapped.

**Reply by laeubi** [reply]
Yes I think this could be a good place to handle this, please note that you probably want to have a similar patch for Tycho as well.
